### PR TITLE
Make serialization methods consistent

### LINKED ISSUE #661: Make serialization methods consistent
There are currently 3 types of (de)serialization functions:

1. `fn deserialize(bytes: <Something>::Serialization)` which is used for Scalar, Field, and Signature types and wrappers
1. `fn deserialize(bytes: &<Something>::Serialization)` same as the previous, but with a reference
1. `fn deserialize(bytes: &[u8])` which is used for all other structs

The obvious inconsistency is between 1 and 2. I was changing all of them to take a reference (option 2) but thinking about it, I wonder if we should make them all follow option 3.

The `<Something>::Serialization` seems like an implementation detail which users shouldn't need to handle. Currently, to deserialize e.g. an identifier you need something like

```rust
        frost_core::Identifier::<C>::deserialize(
            &bytes
                .try_into()
                .map_err(|_| frost_core::Error::<C>::DeserializationError)?,
        )
```

which is super annoying. The user needs to lookup what the `<<C::Group as Group>::Field as Field>::Serialization` type is, and figure out that they have to use `TryFrom<Vec<u8>>`.

If we changed to option 3 in the public API (while still using `<<C::Group as Group>::Field as Field>::Serialization`  internally) then they just have to pass a `&[u8]` which is more straightforward.

### PR DESCRIPTION
**Pull request by conradoplg**
Depends on #671 

Closes #661

### CODE REVIEW

File: frost-core/CHANGELOG.md [hunk 1]
```diff
@@ -17,6 +17,18 @@ Entries are listed in reverse chronological order.
     `VerifiableSecretSharingCommitment::serialize()`,
     `NonceCommitment::serialize()`, `Signature::serialize()`,
     `VerifyingKey::serialize()` can now all return an error.
+* Changed the `serialize()` and `deserialize()` methods of all Scalar- and
+  Element-wrapping structs; instead of taking or returning a
+  `Field::Serialization` or `Element::Serialization` traits (which are usually
```
Comment (natalieesk) [inline frost-core/CHANGELOG.md:22]: Typo, should be "trait" not "traits"

Comment (conradoplg) [reply frost-core/CHANGELOG.md:22]:
```suggestion
  `Field::Serialization` or `Element::Serialization` trait (which are usually
```

Comment (conradoplg) [reply frost-core/CHANGELOG.md]: Fixed

**Review by natalieesk — APPROVED** [review]
Looks good

**Pull request merged**